### ISSUE TITLE
Do we need Pub/Sub in Ember

### ISSUE THREAD

**Issue by MiguelMadero** [emberjs/rfcs#19]
This might turn into its own RFC, but I'm branching off this discussion from a comment on [The Road to 2.0](https://github.com/emberjs/rfcs/pull/15#issuecomment-62368483). 

From @kristianmandrup 
> A better alternative might be simple publish/subcribe from within these lifecycle events to minimize direct coupling.

From @MiguelMadero
> I'm a afraid of global pub/sub for client side applications. I feel It's often abused and it's hard to follow. In the backend, it makes a bit more sense, since it's used to communicate across domain concerns, between processes that are already isolated. Within an app, I see less need for that. I like the explicit interfaces of components, they can communicate with their consumer via actions. Also using `services` and manipulating "shared" models they have access to the rest of the world, extending their list of collaborators and it results in more explicit interfaces, as opposed to the loosey goosey nature of pub/sub.

Pub/sub today is certainly possible by using something like [amplify](http://amplifyjs.com/) for pub/sub. 

From @kristianmandrup 

> Sometimes you just want to broadcast some action globally (within component scope) and have any child component subscribing to this broadcast act on it (perhaps namespaced broadcast like "post:expand"). Same goes the other way (broadcast bubbling up parent component hierarchy: 
"comment:removed" --> higher lv component notices no more comments (shown?) and broadcasts "comments:empty") and some component in the side or navbar showing "number of comments in view", receives this notice and removes itself...

comment:removed and comments:empty I think it could be a data concern. Removing it from the array should notify all of the subscribers. 
post:expand, I usually expand/collapse within the same component so there's no need for external communication. If it's done externally it seems so specific that I think only one component will care about it (only one subscriber) and quite likely one place to do it (one publisher). Are this two components already related? Is the publisher the parent of the consumer? 
If this is a more broad concern, is it possible that we can do it via the router and transitions? 
Could we rely on models and services to do some of this and have more explicit interfaces? 

This is not the first time this topic comes up and I think it merits some analysis.

**Comment by mmun** [comment]
There hasn't been any discussion here so I'm closing the issue. Please reopen on http://discuss.emberjs.com.

**Comment by MiguelMadero** [comment]
Thanks. I mainly wanted to move an off-topic discussion off from RFC #15. I'm not particularly in favor of pub/sub, but @kristianmandrup might want to restart the conversation on discuss.
